Automattic/woocommerce-payments
# Add platform checkout iframe and listen to "redirect" JS message

#### Changes proposed in this Pull Request









Adds an iframe loading the platform checkout "SMS OTP" page. Upon successful authentication, initiate a platform checkout session and redirect to platform checkout.

No automated tests added as this implementation is rather unlikely to remain an iframe.

![Markup on 2021-12-08 at 18:27:13](https://user-images.githubusercontent.com/1759681/145254922-bb8c8d19-a800-4f96-a36c-98dee934e61e.png)

#### Testing instructions





1. Make sure you have a platform checkout server set up.
   * There is a new const introduced by this PR: `PLATFORM_CHECKOUT_FRONTEND_HOST` which should point at your platform checkout's URL.
2. Make sure you have a SMS OTP page on your platform checkout site.
3. Go to either Cart or Checkout on your WCPay site and test! Upon successful SMS OTP validation you should be automatically redirected to platform checkout.

-------------------


**Post merge**
